### PR TITLE
feat(integrationtest): Add happyflow integration test

### PR DESCRIPTION
**Pull request by richardwerkman**
I added a integrationtest project that runs the stryker source code on a small project. Right now it creates 15 mutants of different kinds and checks if the status code is correct. Therefore this PR depends on #154. The testrun takes around 50 seconds on my machine. If this is too long we can change the exampleproject to be smaller. We should add this test to run on the build server.

### CODE REVIEW

**Comment by nicojs** [comment]
Great work! Love to see some integration tests. These should indeed run in the CI build (and when we publish a new version).

Is there a technical reason to keep the solution files separated? Seems strange to me. Having them all in 1 solution seems easier to manage.

Can we validate that reverting #157 results in breaking these tests?

**Comment by richardwerkman** [comment]
@nicojs 
> Is there a technical reason to keep the solution files separated? Seems strange to me. Having them all in 1 solution seems easier to manage.

I created the new solution because the integrationtest takes quite long. This makes running the unit tests using "run all tests" inside visual studio very slow. This would also affect live unit testing. For me personally the integration test doesn't have to run while developing. I just want it to run when creating a PR to see if the happy flow still works.

> Can we validate that reverting #157 results in breaking these tests?

Not yet, I assert that the exit code is correct after the happy flow. Right now the PR (#154) that sets this exit code is not merged yet. So we should wait for that first.

**Comment by richardwerkman** [comment]
closes #156

**Comment by richardwerkman** [comment]
@nicojs the integration test now fails when an exception occurs while running the program. But I noticed the azure pipeline job now almost takes 40 minutes. Is this okay? The appveyor job only took a few minutes.

File: integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject.XUnit/DymmyBooleanTest.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace ExampleProject.XUnit
+{
+    public class DymmyBooleanTest
```
Comment (Mobrockers) [inline integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject.XUnit/DymmyBooleanTest.cs]: Class and file name are incorrect

File: integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject.XUnit/DummyCalcTest.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using Xunit;
+
+namespace ExampleProject.XUnit
+{
+    public class DummyCalcTest
```
Comment (Mobrockers) [inline integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject.XUnit/DummyCalcTest.cs]: Are these tests commented out on purpose?

File: integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject.XUnit/Endlesslooptest.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace ExampleProject.XUnit
+{
+    public class Endlesslooptest
```
Comment (Mobrockers) [inline integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject.XUnit/Endlesslooptest.cs]: Endlesslooptest should be EndlessLoopTest

File: integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject/Endlessloop.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ExampleProject
+{
+    public class Endlessloop
```
Comment (Mobrockers) [inline integrationtests/IntegrationTests/Stryker.Core.IntegrationTest/ExampleProject/ExampleProject/Endlessloop.cs]: EndlessLoop

**Comment by simondel** [comment]
It shouldn't take that long. The build in my previous comment runs the exact same project, only then from the actual NuGet package

**Pull request merged**